sdk/streamlib-python-wheel/tests/test_cli.py

```
@@ -0,0 +1,393 @@
+    with pytest.raises(cli.AppLaunchError) as resolution_failure:
+        cli.resolve_app_entry_file("dev", tmp_path, None)
+
+    message = str(resolution_failure.value)
+    assert "app.py" in message, "the error must name the convention"
+    assert str(tmp_path) in message, "the error must name the anchor it searched"
+    assert "streamlib dev" in message, "the error must name the verb the user typed"
+    assert "-f " in message, "the error must offer the `-f` escape hatch"
+
+
+def test_a_missing_explicit_entry_names_the_path_it_tried(tmp_path: Path):
+    with pytest.raises(cli.AppLaunchError, match="gone.py"):
```

Ruff (0.16.1)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)